### PR TITLE
gui: Generate version-specific documentation link URLs

### LINKED ISSUE #7108: Certificate for relaysrv with better naming
### Include required information

Please be sure to include at least:
 - what problem your new feature would solve
If an AV-software checks the validity of SSL-connections the strelaysrv gets catched because of a selfsign certificate.
If a person checks that, the organisation name in the certificate is strange and it is not obvious that it belongs to syncthing.
I blocked the connections first because I didn't trust the certificate because of the long expire time.
 - how or why you think it is generally useful (i.e., not just for you)
A organisation field filled with O:"Syncting" or CN: "Syncting relay server" would make the certificate much more trustworthy and a person would recognize it. 
 - what alternatives or workarounds you considered
A dialog on the first startup that encourages you to install the certificate.

I attached a screenshot of my av-software warning me about the connection (its in german so sorry for that)
image

### LINKED ISSUE #8107: gui: Links to documentation should point to matching version
Original discussion on the forum: https://forum.syncthing.net/t/version-specific-documentation-proposal/17858

The documentation links in some places of the Web GUI currently point at the single URL `https://docs.syncthing.net/...`.  That is a moving target however, documenting stuff that might not yet be found in the currently running version or missing features that have been removed / reworked in a subsequent release and already was removed documentation.  The envisioned solution consists of these base components:

1. [x] Create an archive of the built HTML result for every version tagged on [the docs source repository](https://github.com/syncthing/docs/).
2. [x] Serve them (statically) under some version-specific URL, e.g. `https://docs.syncthing.net/v1.18.6/...` or `https://docs.syncthing.net/...?version=v1.18.6`.
3. [ ] Alter the links contained in the Web GUI to include (part of) the version string for the running instance.
4. [ ] Add fall-back logic on the server side to redirect any unavailable version URL to the generic, up-to-date "main" version as it is now.
5. [x] Keep a JSON file on the server listing all available deployed documentation version builds.
6. [ ] Embed the version tag in built HTML docs to easily see what one is looking at.

An optional future refinement would be to add a version selector on the docs site to quickly switch between versions, e.g. when comparing API evolution.  But that is out of scope for this request.

### PR DESCRIPTION
**Pull request by acolomb**
One puzzle piece to fix #8107.  Should be merged only when the other pieces are deployed.

Replace all HTML references to `https://docs.syncthing.net` with a function call to return a version-specific URL.  Use the currently running version string up to the first dash (-) character as a subdirectory name on the docs site.

Fall back to the generic base URL on error.

### CODE REVIEW

File: gui/default/index.html [hunk 1]
```diff
@@ -432,7 +432,7 @@ <h4 class="panel-title">
                             <i class="small">
                               <span translate style="white-space: normal;">Altered by ignoring deletes.</span>
                               <br>
-                              <a href="https://docs.syncthing.net/advanced/folder-ignoredelete" target="_blank"><span class="fas fa-question-circle"></span>&nbsp;<span translate>Help</span></a>
+                              <a href="{{docsURL('advanced/folder-ignoredelete" target="_blank"><span class="fas fa-question-circle"></span>&nbsp;<span translate>Help</span></a>
```
Comment (calmh) [inline gui/default/index.html:435]:
```suggestion
                              <a href="{{docsURL('advanced/folder-ignoredelete')}}" target="_blank"><span class="fas fa-question-circle"></span>&nbsp;<span translate>Help</span></a>
```

File: gui/default/syncthing/core/syncthingController.js [hunk 2]
```diff
@@ -2887,6 +2887,34 @@ angular.module('syncthing.core')
...
+                    url += '/' + path.slice(0, hash);
+                    url += '?version=' + $scope.versionBase();
+                    url += path.slice(hash);
+                } else {
+                    url += '/' + path;
+                    url += '?version=' + $scope.versionBase();
+                }
+            }
+            return url;
+        };
```
Comment (calmh) [inline gui/default/syncthing/core/syncthingController.js:2916]: I realized something interesting about our docs serving at the moment, in that most of the old URLs here ended with `.html` but one didn't, and the one that didn't still worked because apparently Netlify does magic and adds it internally instead of 404:ing. That makes the above code (which omits the `.html`) work, and means we can lay the foundation for a switch to generating nicer docs urls with `make dirhtml` without breaking everything...

**Review by calmh — DISMISSED** [review]

File: gui/default/syncthing/core/syncthingController.js (hunk 2 above)
Comment (calmh) [reply gui/default/syncthing/core/syncthingController.js:2916]: It is however irreversible in the sense that we'll then have versions out there referring to both /foo and /foo.html and we'll probably need to handle both variants server side for a long while or risk breaking some subset of users.

**Comment by tomasz1986** [comment]
Is there any indication on the Docs page that we're actually browsing a version-specific site rather than the main one? If not, then I'd strongly suggest to either have the version number written somewhere visible on the screen or, if possible, to have a dropdown menu allowing the user to select between different versions.

Please correct me if I'm wrong, but at the moment it seems that it is only the URL that is clearly version-specific. If that is the case, then I'd say it's really not enough, because the vast majority of users will simply fail to notice it, and then they'll wonder what is going on (especially since up to now the GUI has used to simply link to the main Docs site). Also, on small screens, that part of the URL may very well be hidden unless the user clicks the address bar and scrolls it.

In other words, I think there are two potential problems. Firstly, the user will likely be unaware that they're browsing version-specific documentation. Secondly, there's no user-friendly way of switching to the current or between different Docs versions.

**Comment by calmh** [comment]
Why would they wonder what's going on? What scenario are you envisioning causing the confusion?

**Comment by tomasz1986** [comment]
Well, I, for one, am used to be able to quickly jump to the Docs from the Web GUI. The current behaviour is to open the main Docs site. The new behaviour is going to open a verion-specific site instead. I'd say this is highly unexpected to everyone who's actually been using the link to open and browse the Docs.

File: gui/default/syncthing/folder/editFolderModalView.html [hunk 3]
```diff
@@ -164,7 +164,7 @@
               </p>
             </div>
             <hr />
-            <p class="small"><span translate>Quick guide to supported patterns</span> (<a href="https://docs.syncthing.net/users/ignoring.html" target="_blank" translate>full documentation</a>):</p>
+            <p class="small"><span translate>Quick guide to supported patterns</span> (<a href="docsURL('users/ignoring')" target="_blank" translate>full documentation</a>):</p>
```
Comment (calmh) [inline gui/default/syncthing/folder/editFolderModalView.html:167]:
```suggestion
            <p class="small"><span translate>Quick guide to supported patterns</span> (<a href="{{docsURL('users/ignoring')}}" target="_blank" translate>full documentation</a>):</p>
```

**Comment by calmh** [comment]
Your expectation isn't suddenly wrong because of this change, the only difference is that you're not getting documentation for some other random version than the one you're running. I don't see this as a problem.

**Comment by tomasz1986** [comment]
This doesn't change the fact that it's still a significant change of the previous behaviour. My opinion is that sudden changes with no indication of them happening are never good for user experience. As long there's some kind of an indicator (i.e. the banner or similar), I've got no problem with it, but as it is right now, the redirection is done silently, and dissecting the URL seems to be the only way to even know about it.

**Comment by calmh** [comment]
Yeah I disagree on the significance of the change, but I filed https://github.com/syncthing/docs/issues/718 so we don't forget nonetheless.

**Comment by acolomb** [comment]
That was already recorded as one remaining task under #8107. And the suggestion with a version switcher is implemented on a technical level already, just needs some styling now.

I agree that nobody would probably care whether they see the main site or the one specific to their version. The latter is actually meant as an *improvement*. For you @tomasz1986 it won't make a difference since I assume you are always up to date or even running development builds. And actually older versions will not change behavior, it's only versions from when we merge this onwards where there will be a visible effect. Nobody suddenly gets to a different site.

**Comment by acolomb** [comment]
Sorry @calmh about the syntax errors, must have been low on concentration while fixing the merge conflicts... Thanks for fixing them.

File: gui/default/syncthing/core/syncthingController.js (hunk 2 above)
Comment (acolomb) [reply gui/default/syncthing/core/syncthingController.js:2916]: I agree, since there have been many versions now with mixed variants, we're bound to keep the fall-back behavior on the server.  Should be relatively easy to do though on any hosting?  Just needs to be remembered.  And in the end, having some docs links from ancient versions stop working someday isn't the end of the world.  The internal links *within* the docs are correct anyway.  So that's a go in leaving out `.html` from now on?

**Review by calmh — APPROVED** [review]

**Comment by tomasz1986** [comment]
Just a quick comment, but I believe the main footer link to the Docs was omitted in this one. Was it done on purpose or just by mistake?

**Comment by acolomb** [comment]
That was on purpose. To have a link to the "project wide" site just like the forum and other web links. Makes sense to me at least, but it could be argued...

**Comment by tomasz1986** [comment]
No strong opinion here, but to tell the truth, I did expect the link to be version specific, hence the surprise 😉.
